File: leetcode/283. Move Zeroes.cpp
// Copyright 2016 Qi Wang
// Date: 2016-10-03
class Solution {
 public:
  void moveZeroes(vector<int>& nums) {
    int nonZeroIdx = 0;
    for (size_t i = 0; i < nums.size(); ++i)
      if (nums[i] != 0) swap(nums[nonZeroIdx++], nums[i]);
    memset(nums.data() + nonZeroIdx, 0,
        sizeof(int) * (nums.size() - nonZeroIdx));
  }
};

File: leetcode/283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        non_zero_idx = 0
        for num in nums:
            if num != 0:
                nums[non_zero_idx] = num
                non_zero_idx = non_zero_idx + 1
        zero_idx = non_zero_idx
        while zero_idx < len(nums):
            nums[zero_idx] = 0
            zero_idx = zero_idx + 1